### PR TITLE
patch eio to set cookies in node

### PR DESCRIPTION
**Pull request by ozsay**

### CODE REVIEW

File: packages/engine.io-client/src/eio-node-patch.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    return req;
+  };
+
+  Request.prototype.onLoad = function () {
+    const content = this.xhr.getResponseHeader('Set-Cookie');
+
+    if (content) {
+      const cookies = _.map(content, c => Cookie.parse(c));
+      const cookiesString = _.map(cookies, c => c.cookieString());
+      const cookieHeader = _.join(_.reverse(cookiesString), '; ');
```
Comment (sloops77) [inline packages/engine.io-client/src/eio-node-patch.js]: can use a chain

File: packages/engine.io-client/src/eio-node-patch.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import _ from 'lodash';
```
Comment (sloops77) [inline packages/engine.io-client/src/eio-node-patch.js]: please use function imports here, for consistency reasons

File: packages/engine.io-client/src/eio-node-patch.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import XHR, { Request } from 'engine.io-client/lib/transports/polling-xhr';
+
+if (_.get(process, 'versions.node')) {
+  const onLoad = Request.prototype.onLoad;
+  const request = XHR.prototype.request;
+
+  XHR.prototype.request = function (opts) {
+    const req = request.call(this, opts);
+
+    req.transport = this;
```
Comment (sloops77) [inline packages/engine.io-client/src/eio-node-patch.js]: can you add a comment about this. seems super weird

Comment (ozsay) [inline packages/engine.io-client/src/eio-node-patch.js]: this is why it's monkey patching :)

**Comment by sloops77** [comment]
How tested?

**Pull request merged**